packages/synapse-sdk/src/storage/context.ts

```diff
@@ -683,34 +685,70 @@ export class StorageContext {
 
   /**
    * Static method to perform preflight checks for an upload
-   * @param size - The size of data to upload in bytes
-   * @param withCDN - Whether CDN is enabled
+   * Composes cost calculation from WarmStorageService and readiness checks from PaymentsService
    * @param warmStorageService - WarmStorageService instance
    * @param paymentsService - PaymentsService instance
+   * @param size - The size of data to upload in bytes
    * @returns Preflight check results without provider/dataSet specifics
    */
   static async performPreflightCheck(
     warmStorageService: WarmStorageService,
     paymentsService: PaymentsService,
-    size: number,
-    withCDN: boolean
+    size: number
   ): Promise<PreflightInfo> {
     // Validate size before proceeding
     StorageContext.validateRawSize(size, 'preflightUpload')
 
-    // Check allowances and get costs in a single call
-    const allowanceCheck = await warmStorageService.checkAllowanceForStorage(size, withCDN, paymentsService)
+    // Calculate upload cost
+    const cost = await warmStorageService.calculateUploadCost(size)
 
-    // Return preflight info
-    return {
-      estimatedCost: {
-        perEpoch: allowanceCheck.costs.perEpoch,
-        perDay: allowanceCheck.costs.perDay,
-        perMonth: allowanceCheck.costs.perMonth,
+    // Calculate rate per epoch from floor-adjusted monthly price
+    const pricing = await warmStorageService.getServicePrice()
+    const ratePerEpoch = cost.withFloorPerMonth / pricing.epochsPerMonth
+
+    // Calculate lockup requirements
+    const lockupEpochs = BigInt(TIME_CONSTANTS.DEFAULT_LOCKUP_DAYS * TIME_CONSTANTS.EPOCHS_PER_DAY)
+    const lockupNeeded = ratePerEpoch * lockupEpochs
```

Comment:
really dont like this per month thing its inaccurate and forces us to do this back and forth stuff epoch<>month exchange.

also we just called getServicePrice in calculateUploadCost and we call it again just to get

https://github.com/FilOzone/filecoin-services/blob/980e25aeb920bb894eab71a3dba9e8c9bccddcae/service_contracts/src/FilecoinWarmStorageService.sol#L200

Reply:
point taken on the double call to `getServicePrice`, could just do that once.

> really dont like this per month thing

but the problem is twofold:

1. the contract defines pricing per month
2. downscaling to per-epoch loses a lot of precision (a surprising amount since we're dealing with fairly low numbers in the first place)

the logic here is intentionally keeping things at per-month for as long as possible to avoid the early precision loss of going per-epoch and then back again.